### PR TITLE
field matching to XML. updated history_event_reason

### PR DESCRIPTION
**Pull request by PatrikLundell**

### CODE REVIEW

File: df.history.xml [hunk 1]
```diff
@@ -824,94 +824,112 @@
...
-
-        <enum-item name='artifact_is_heirloom_of_family_hfid'/>
-        <enum-item name='cement_bonds_of_friendship'/>
-        <enum-item name='as_a_symbol_of_everlasting_peace'/>
-        <enum-item name='on_a_pilgrimage'/>
-        <enum-item name='gather_information'/>
-        <enum-item name='seek_sanctuary'/>
-        <enum-item name='part_of_trade_negotiation'/>
-        <enum-item name='artifact_is_symbol_of_entity_position'/>
+        <enum-item name='insurrection' caption='for an insurrection'/>
```
Comment (lethosor) [inline df.history.xml]: I'm pretty sure `caption` isn't a recognized attribute, so this won't show up anywhere.
If you want code to be able to access these, a better option would be to use enum-attrs, like this: https://github.com/DFHack/df-structures/blob/8d3191569199b8d65304e688046ea7766c6b3621/df.job-types.xml#L24-L45

**Comment by PatrikLundell** [comment]
The honors mapping was largely done through comparison of XML export and the data in the save exported.

File: df.history.xml [hunk 2]
```diff
@@ -2515,10 +2515,10 @@
...
       <int32_t name='source_site' ref-target='world_site'/>
       <int32_t name='dest_site' ref-target='world_site'/>
-      <int32_t/>
-      <int32_t/>
-      <int32_t/>
-      <int32_t comment='value of the trade?'/>
+      <int32_t name='unk_v47_1' comment='Somehow connected to traded item, but not mat type/index'/>
+      <int32_t name='unk_v47_2' comment='Somehow connected to traded item, but not mat type/index'/>
+      <int32_t name='unk_v47_3' comment='Somehow connected to traded item, but not mat type/index'/>
+      <int32_t name='profit'/>
```
Comment (lethosor) [inline df.history.xml]: @sedb-dfh identified these in https://github.com/DFHack/df-structures/pull/350/files#diff-35ab45624fa78d4a8c071f3c0fee5d76R2282-R2285

Comment (PatrikLundell) [reply df.history.xml]: Good. I've copied those changes.

**Pull request merged**